### DISCUSSION TITLE
Is it even possible to test minified build now?

### DISCUSSION THREAD

**Discussion by dsvoronin** [slackhq/keeper#146] [Q&A] [answered]
Hello community!
After upgrade to AGP 8.x we've noticed this warning:

```
BuildType 'debug' is both debuggable and has 'isMinifyEnabled' set to true.
Debuggable builds are no longer name minified and all code optimizations and obfuscation will be disabled.
```

But it's impossible to run instrumented tests with `debuggable=false`, right?

Am I missing something? Is there a way to test minified and obfuscated builds now?

**Comment by ZacSweers** [comment]
Sorry I'm not seeing how this is related to Keeper?

**Reply by dsvoronin** [reply]
> Sorry I'm not seeing how this is related to Keeper?

I see Keeper purpose as to assist with running UI tests over minified builds.

My assumption was that if Keeper updated to recent AGP, than there is a known workaround

**Reply by ZacSweers** [reply]
You're not really being very clear, I don't understand what keeper's relationship to this is. We have pretty clear docs on how to set up a test build type for use with keeper.

**Reply by dsvoronin** [reply]
> You're not really being very clear, I don't understand what keeper's relationship to this is. We have pretty clear docs on how to set up a test build type for use with keeper.

Yep, you're right. I'll try to explain more in the answer, as I already found answers.
Hope this research would be helpful here, if you don't think so, feel free to remove the topic

**Comment by dsvoronin** [comment, marked as answer]
So first of all, my assumption was wrong, and instrumented test WORKS over build with  `debuggable=false` just fine. 
If you know any limitations, please let me know.

In case you want to debug failed test on the same build, which is, I believe, a fair requirement, than you have to do couple of tricks. Example:

https://github.com/RedMadRobot/gradle-infrastructure/pull/122/commits/7e70dc2825d9245154b5f014db70bc0972ef3c92
Thanks @osipxd .

You also need to change `BuildConfig.DEBUG` value

Some history:
- minification and obfuscation is not working [since AGP 7.2.0](https://issuetracker.google.com/issues/238655204), is was silently(no release notes mention) disabled as not intended behavior
